drivers/docker/handle.go

```diff
@@ -159,7 +167,13 @@ func (h *taskHandle) Kill(killTimeout time.Duration, signal string) error {
 	// Signal is used to kill the container with the desired signal before
 	// calling StopContainer
 	if signal == "" {
-		err = h.client.StopContainer(h.containerID, uint(killTimeout.Seconds()))
+		// give the context timeout some wiggle room beyond the kill timeout
+		// docker will use, so we can happy path even in the force kill case
+		graciousTimeout := killTimeout + dockerTimeout
+		ctx, cancel := context.WithTimeout(context.Background(), graciousTimeout)
+		defer cancel()
+		apiTimeout := uint(killTimeout.Seconds())
+		err = h.infinityClient.StopContainerWithContext(h.containerID, apiTimeout, ctx)
```

Comment:
This is the major user-facing change, right? We'll want a changelog item for this.

Reply:
Yup!